lib/ifconfig/sunos/ifconfig.rb

```diff
@@ -31,7 +31,8 @@ def initialize(input=nil,netstat=nil,verbose=nil)
         when /\s+ether\s+/im
           @ifaces[iface_name] = EthernetAdapter.new(iface_name,iface,netstat)
         else
-          puts "Unknown Adapter Type: #{iface}" if @verbose
+          # fallback for virtual nics like bge:0
+          @ifaces[iface_name] = EthernetAdapter.new(iface_name,iface,netstat)
```

Comment:
It's inappropriate to use EthernetAdapter here because EthernetAdapters must have a MAC.  Is a "virtual nic" the same thing as what other operating systems call an alias?  Or is it an SR-IOV virtual function?  Or something else?